chartmogul/resource.py

```diff
@@ -198,14 +200,16 @@ def _validate_arguments(cls, method, kwargs):
             )
 
     @classmethod
-    def _method(cls, method, http_verb, path=None):
+    def _method(callerClass, method, http_verb, path=None, useCallerClass=False):
         @classmethod
-        def fc(cls, config, **kwargs):
+        def fc(calleeClass, config, **kwargs):
             if config is None or not isinstance(config, Config):
                 raise ConfigurationError(
                     "First argument should be" " instance of chartmogul.Config class!"
                 )
 
+            cls = callerClass if useCallerClass else calleeClass
```

Comment:
This was done because doing this below without the modification will result in an error:

```python
Customer.contacts = Contact._method('all', 'get', '/customers/{uuid}/contacts')
```

The reason is, when `._request` is run, it will use the `Customer` class instead. In normal circumstances, it should be run using the same class, but in this case, `._method` is called by a different class. Therefore, this was needed. If there's a better way to do this, let me know